scipy/stats/tests/test_morestats.py

```diff
@@ -1987,6 +1987,79 @@ def test_circfuncs_unit8(self):
         assert_allclose(stats.circstd(x, high=180), 20.91551378, rtol=1e-7)
 
 
+class TestQuartileCoeffOfDispersion(object):
+
+    def test_simple_flattened(self):
+        arr = [2, 4, 6, 8, 10, 12, 14, 16]
+
+        # Q1 = 4, Q2 = 12 -> dispersion = (12-4)/(12+4) = 8/16 = 0.5
+        assert_equal(stats.quartile_coeff_dispersion(arr), 0.5)
+
+        arr = [1.6, 2.1, 2.3, 2.4, 2.6, 2.9, 2.98, 3]
+
+        # Q1 = 2.1, Q2 = 2.9 -> dispersion = (2.9-2.1)/(2.9+2.1) = 0.8/5 ~= 0.16
+        assert_almost_equal(stats.quartile_coeff_dispersion(arr), 0.16)
+
+    def test_simple_many(self):
+        ref = [2, 4, 6, 8, 10, 12, 14, 16]
+        arr = np.array([ref] * len(ref)).transpose()
+
+        # Same as test_simple_flattened but with arr as repeated columns, expecting len(arr) 0.5 dispersion values
+        assert_array_equal(stats.quartile_coeff_dispersion(arr, axis=0), [0.5] * len(ref))
+        assert_array_equal(stats.quartile_coeff_dispersion(arr, ), 0.5)
+
+        # Aggregate over row, so each row has the same value -> Q2 = 0 -> dispersion = 0
+        assert_array_equal(stats.quartile_coeff_dispersion(arr, axis=1), [0.0] * len(ref))
+
+    def test_q1_q2(self):
+        arr = [2, 4, 6, 8, 10, 12, 14, 16]
+
+        # Q1 (0.25) = 4, Q2 (0.5) = 8 -> dispersion = (8-4)/(8+4) = 4/12 = 1/3
+        assert_equal(stats.quartile_coeff_dispersion(arr, q=(0.25, 0.5)), 1/3)
+
+        ref = [2, 4, 6, 8, 10, 12, 14, 16]
+        arr = np.array([ref] * len(ref)).transpose()
+
+        # Same as above but with arr as repeated columns, expecting len(arr) 1/3 dispersion values
+        assert_array_almost_equal(stats.quartile_coeff_dispersion(arr, q=(0.25, 0.5), axis=0), [1/3] * len(ref))
+
+        arr = [1.6, 2.1, 2.3, 2.4, 2.6, 2.9, 2.98, 3]
+
+        # Q1 (0.25) = 2.1, Q2 (0.5) = 2.4 -> dispersion = (2.4-2.1)/(2.4+2.1) = 0.3/4.5 ~= 0.0666
+        assert_almost_equal(stats.quartile_coeff_dispersion(arr, q=(0.25, 0.5)), (2.4 - 2.1)/(2.4 + 2.1))
```

Comment:
It's OK to leave these in, but I don't know that the behavior on multidimensional arrays needs to be checked for all values of the `qs`, as there's no reason to suspect that changing these would break the way the functions works on arrays.

Reply:
That's right, should I remove? It may add some granularity to the UT's - maybe better to change to different dispersion targets?

Reply:
I'm not sure. I'd just suggest that you think about the purpose of each test. 

For example, you have already tested above that
```python3
assert_equal(stats.quartile_coeff_dispersion(arr, q=(0.25, 0.5)), 1/3)
```
and in `test_simple_many`:
```python3
assert_array_equal(stats.quartile_coeff_dispersion(arr, axis=0), [0.5] * len(ref))
```
Based on the code, I don't think there is any reason to suspect that specifying `q`  is going to break the behavior on arrays, but perhaps it doesn't hurt to check the combination of array input and custom `q`s:
```python3
assert_array_almost_equal(stats.quartile_coeff_dispersion(arr, q=(0.25, 0.5), axis=0), [1/3] * len(ref))
```
But if you've done that once, then I'm not sure that
```python3
assert_array_almost_equal(stats.quartile_coeff_dispersion(arr, q=(0.5, 0.75), axis=0), [1/5] * len(ref))
```
tests anything new, unless we suspect that NumPy might make arithmetical mistakes.

Some redundancy is OK - it's better than forgetting something. But if you're going to be testing the same sorts of things for different combinations of input array values, input array sizes, custom quantiles, and `axis` arguments, then I would prefer that you [@parametrize](https://docs.pytest.org/en/stable/parametrize.html) the tests instead of copy-pasting. That would make the test code shorter, which makes it and easier to confirm that it covers the function adequately. 

Also, I appreciate your comments stating where values come from, e.g.
```
# Q1 (0.5) = 8, Q2 (0.75) = 12 -> dispersion = (12-8)/(12+8) = 4/20 = 1/5
```
but one thing that would help is comments indicating the purpose of each part of the test. For example, is the purpose of the test exactly the same as previous tests but with different numbers, or are you testing a different feature of the code?